Feature/consolidate more settings into base

bermudezjd opened the pull request:
Second pass at settings consolidation, moving database and cache settings into base.  Removed a lot of unused files.  Also introduces aws settings and requirements files and essentially deprecates other environment files by having them import the aws version.

@eparker71 
@elliottyates 
@rascalking 
@douglashall

eparker71 commented on icommons_ext_tools/settings/base.py:
```diff
@@ -77,9 +158,23 @@
-# DATABASES are defined in individual environment settings
+# DATABASES
+
+DATABASES = {
```
databases is defined twice, it's also on line 81

bermudezjd commented:
OK @eparker71 , I fixed that db change and made some other updates in line with the lti_emailer settings I've modified.

rascalking commented on icommons_ext_tools/settings/base.py:
```diff
@@ -58,256 +41,150 @@
+    'icommons_ui',
+    'qualtrics_link',
+    'crispy_forms',
 )
```
I thought we were keeping canvas_course_wizard in lti_tools & ext_tools until after the fall freeze, no?

bermudezjd replied:
This PR was based on @eparker71's PR #93 , which had removed the wizard and corresponding config.  Looks like perhaps that was closed... I had originally thought that the wizard was getting removed after the most recent demo, but I can add all of that config back if needed.

rascalking replied:
Yeah, I missed out on the discussions, but I believe the final decision was to basically revert that until after fall freeze.  @djbrooke @sapnamysore @douglashall can you guys confirm?

djbrooke replied:
That's correct - there's some additional info here:

https://jira.huit.harvard.edu/browse/TLT-1808?focusedCommentId=55422&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-55422

bermudezjd replied:
Awesome... so I'll go through and revert those changes then.  Thanks for the clarification @djbrooke

rascalking commented on manage.py:
```diff
@@ -6,7 +6,7 @@
         os.environ['DJANGO_SETTINGS_MODULE'] = 'icommons_ext_tools.settings.test_settings'
     else:
-        os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'icommons_ext_tools.settings.local')
+        os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'icommons_ext_tools.settings.aws')
```
I think this still wants to be `.local`, no?

bermudezjd replied:
I think it would be more beneficial if it defaulted to AWS as the wsgi file does so that we don't have to set the `DJANGO_SETTINGS_MODULE` in every management command we want to run during deployment.  I suppose once _every_ project is converted to using AWS

rascalking replied:
Yeah, my major worry here was having to remember to pass it all the time when running under vagrant.

bermudezjd replied:
OK, I'll add it to the vagrant setup then as part of this PR and let you know when it's ready to go.

bermudezjd commented:
@rascalking I added the course site wizard back into the project and updated the provisioning to use local settings.  let me know if that's good for you.  I _think_ you can just do a `vagrant provision` to get the necessary updates.

rascalking commented:
Thanks.  I think it only needs a `pip install --upgrade -r icommons_ext_tools/requirements/local.txt`.

Looks good, btw.

bermudezjd commented:
OK.  I'm pretty sure you'll have to reprovision your vagrant box in order to get the change that sets `DJANGO_SETTINGS_MODULE` to be `icommons_ext_tools.settings.local` when you're working within the virtualenv though.  In any case, is that a :+1: ?

rascalking commented:
Ah, I see what you mean...thought you were just talking about pulling in the new dependencies.  And yeah, :+1: .
